modules/memberships/class-jetpack-memberships.php

```diff
@@ -0,0 +1,289 @@
+		/*
+		 * PLAN data structure.
+		 */
+		$capabilities = array(
+			'edit_post'          => 'edit_posts',
+			'read_post'          => 'read_private_posts',
+			'delete_post'        => 'delete_posts',
+			'edit_posts'         => 'edit_posts',
+			'edit_others_posts'  => 'edit_others_posts',
+			'publish_posts'      => 'publish_posts',
+			'read_private_posts' => 'read_private_posts',
+		);
+		$order_args   = array(
+			'label'               => esc_html__( 'Plan', 'jetpack' ),
+			'description'         => esc_html__( 'Memberships plans', 'jetpack' ),
+			'supports'            => array( 'title', 'custom-fields', 'content' ),
+			'hierarchical'        => false,
+			'public'              => false,
+			'show_ui'             => false,
+			'show_in_menu'        => false,
+			'show_in_admin_bar'   => false,
+			'show_in_nav_menus'   => false,
+			'can_export'          => true,
+			'has_archive'         => false,
+			'exclude_from_search' => true,
+			'publicly_queryable'  => false,
+			'rewrite'             => false,
+			'capabilities'        => $capabilities,
+			'show_in_rest'        => false,
+		);
+		register_post_type( self::$post_type_plan, $order_args );
+	}
+
+	/**
+	 * Allows custom post types to be used by REST API.
+	 *
+	 * @param array $post_types - other post types.
+	 *
+	 * @see hook 'rest_api_allowed_post_types'
+	 * @return array
+	 */
+	public function allow_rest_api_types( $post_types ) {
+		$post_types[] = self::$post_type_plan;
+
+		return $post_types;
+	}
+
+	/**
+	 * Allows custom meta fields to sync.
+	 *
+	 * @param array $post_meta - previously changet post meta.
+	 *
+	 * @return array
+	 */
+	public function allow_sync_post_meta( $post_meta ) {
+		$meta_keys = array_map(
+			array( $this, 'return_meta' ),
+			$this->get_plan_property_mapping()
+		);
+		return array_merge( $post_meta, array_values( $meta_keys ) );
+	}
+
+	/**
+	 * This returns meta attribute of passet array.
+	 * Used for array functions.
+	 *
+	 * @param array $map - stuff.
+	 *
+	 * @return mixed
+	 */
+	public function return_meta( $map ) {
+		return $map['meta'];
+	}
+	/**
+	 * Callback that parses the membership purchase shortcode.
+	 *
+	 * @param array $attrs - attributes in the shortcode. `id` here is the CPT id of the plan.
+	 *
+	 * @return string|void
+	 */
+	public function render_button( $attrs ) {
+		Jetpack_Gutenberg::load_assets_as_required( self::$button_block_name, array( 'thickbox', 'wp-polyfill' ) );
+
+		if ( empty( $attrs['planId'] ) ) {
+			return;
+		}
+		$id      = intval( $attrs['planId'] );
+		$product = get_post( $id );
+		if ( ! $product || is_wp_error( $product ) ) {
+			return;
+		}
+		if ( $product->post_type !== self::$post_type_plan || 'publish' !== $product->post_status ) {
+			return;
+		}
+
+		$data = array(
+			'blog_id'      => self::get_blog_id(),
+			'id'           => $id,
+			'button_label' => __( 'Your contribution', 'jetpack' ),
+			'powered_text' => __( 'Powered by WordPress.com', 'jetpack' ),
```

Comment:
Should it say "Powered by Jetpack" for Jetpack sites?

Reply:
No, it should not. This is for the purpose of alerting subscribers that they will need wordpress.com account to facilitate the payment.